--- a/platform/linuxbsd/wayland/wayland_thread.cpp
+++ b/platform/linuxbsd/wayland/wayland_thread.cpp
@@ -324,14 +324,19 @@
 	ERR_FAIL_NULL(wayland_thread);
 
 	Key last_key = Key::NONE;
-	xkb_compose_status compose_status = xkb_compose_state_get_status(p_ss->xkb_compose_state);
+	xkb_compose_status compose_status = XKB_COMPOSE_NOTHING;
+	if (p_ss->xkb_compose_state) {
+		compose_status = xkb_compose_state_get_status(p_ss->xkb_compose_state);
+	}
 
 	if (p_pressed) {
 		xkb_keysym_t keysym = xkb_state_key_get_one_sym(p_ss->xkb_state, p_xkb_keycode);
-		xkb_compose_feed_result compose_result = xkb_compose_state_feed(p_ss->xkb_compose_state, keysym);
-		compose_status = xkb_compose_state_get_status(p_ss->xkb_compose_state);
+		xkb_compose_feed_result compose_result = XKB_COMPOSE_FEED_IGNORED;
+		if (p_ss->xkb_compose_state) {
+			compose_result = xkb_compose_state_feed(p_ss->xkb_compose_state, keysym);
+			compose_status = xkb_compose_state_get_status(p_ss->xkb_compose_state);
+		}
 
 		if (compose_result == XKB_COMPOSE_FEED_ACCEPTED && compose_status == XKB_COMPOSE_COMPOSED) {
 			// We need to generate multiple key events to report the composed result, One
 			// per character.
 			char str_xkb[256] = {};
@@ -2282,8 +2287,13 @@
 	}
 	ss->xkb_compose_table = xkb_compose_table_new_from_locale(ss->xkb_context, locale, XKB_COMPOSE_COMPILE_NO_FLAGS);
 
-	xkb_compose_state_unref(ss->xkb_compose_state);
-	ss->xkb_compose_state = xkb_compose_state_new(ss->xkb_compose_table, XKB_COMPOSE_STATE_NO_FLAGS);
+	if (ss->xkb_compose_state) {
+		xkb_compose_state_unref(ss->xkb_compose_state);
+	}
+	ss->xkb_compose_state = nullptr;
+	if (ss->xkb_compose_table) {
+		ss->xkb_compose_state = xkb_compose_state_new(ss->xkb_compose_table, XKB_COMPOSE_STATE_NO_FLAGS);
+	}
 
 	xkb_state_update_mask(ss->xkb_state, ss->mods_depressed, ss->mods_latched, ss->mods_locked, 0, 0, ss->current_layout_index);
 }
