--- a/IGC/CMakeLists.txt
+++ a/IGC/CMakeLists.txt
@@ -522,7 +522,7 @@ if(MSVC)
       GROUP NAME Optimization       "/O[d12x]"
       GROUP NAME RuntimeLibrary     "/M[TD]d?"
       GROUP NAME Rtti               "/GR-?"
-      GROUP NAME TreatWarnAsError   "/WX-?"
+      #GROUP NAME TreatWarnAsError   "/WX-?"
 
       ALIAS AdditionalOptions          "/w14062" "/Wv:18"
       ALIAS CallingConvCdecl           "/Gd"
@@ -551,8 +551,8 @@ if(MSVC)
       ALIAS RtMultiThreadedSharedDebug "/MDd"
       ALIAS RttiDisabled               "/GR-"
       ALIAS RttiEnabled                "/GR"
-      ALIAS TreatWarnAsErrorDisabled   "/WX-"
-      ALIAS TreatWarnAsErrorEnabled    "/WX"
+      #ALIAS TreatWarnAsErrorDisabled   "/WX-"
+      #ALIAS TreatWarnAsErrorEnabled    "/WX"
       ALIAS UninitializedVarWarning    "/w14701"
       ALIAS UnreferencedVariable       "/w14189"
       ALIAS WholeProgramOptimization   "/GL"
@@ -569,7 +569,7 @@ elseif(_igc_compiler_is_gnu)
       GROUP NAME Exceptions         "-f(no-)?exceptions"
       GROUP NAME Optimization       "-O[0-3sg]?"
       GROUP NAME Rtti               "-f(no-)?rtti"
-      GROUP NAME TreatWarnAsError   "-Werror"
+      #GROUP NAME TreatWarnAsError   "-Werror"
 
       ALIAS AdditionalOptions
       ALIAS CallingConvCdecl
@@ -598,8 +598,8 @@ elseif(_igc_compiler_is_gnu)
       ALIAS RtMultiThreadedSharedDebug
       ALIAS RttiDisabled               "-fno-rtti"
       ALIAS RttiEnabled                "-frtti"
-      ALIAS TreatWarnAsErrorDisabled   TreatWarnAsError REMOVE_GROUP
-      ALIAS TreatWarnAsErrorEnabled    "-Werror"
+      #ALIAS TreatWarnAsErrorDisabled   TreatWarnAsError REMOVE_GROUP
+      #ALIAS TreatWarnAsErrorEnabled    "-Werror"
       ALIAS UnreferencedVariable
       ALIAS UninitializedVarWarning
       ALIAS WholeProgramOptimization
@@ -628,7 +628,7 @@ elseif(_igc_compiler_is_clang)
       GROUP NAME Exceptions         "-f(no-)?exceptions"
       GROUP NAME Optimization       "-O([0-3]?|fast|s|z)?"
       GROUP NAME Rtti               "-f(no-)?rtti"
-      GROUP NAME TreatWarnAsError   "-Werror"
+      #GROUP NAME TreatWarnAsError   "-Werror"
 
       ALIAS AdditionalOptions
       ALIAS CallingConvCdecl
@@ -657,8 +657,8 @@ elseif(_igc_compiler_is_clang)
       ALIAS RtMultiThreadedSharedDebug
       ALIAS RttiDisabled               "-fno-rtti"
       ALIAS RttiEnabled                Rtti REMOVE_GROUP
-      ALIAS TreatWarnAsErrorDisabled   TreatWarnAsError REMOVE_GROUP
-      ALIAS TreatWarnAsErrorEnabled    "-Werror"
+      #ALIAS TreatWarnAsErrorDisabled   TreatWarnAsError REMOVE_GROUP
+      #ALIAS TreatWarnAsErrorEnabled    "-Werror"
       ALIAS UnreferencedVariable
       ALIAS UninitializedVarWarning
       ALIAS WholeProgramOptimization
@@ -698,8 +698,8 @@ else()
       ALIAS RtMultiThreadedSharedDebug
       ALIAS RttiDisabled
       ALIAS RttiEnabled
-      ALIAS TreatWarnAsErrorDisabled
-      ALIAS TreatWarnAsErrorEnabled
+      #ALIAS TreatWarnAsErrorDisabled
+      #ALIAS TreatWarnAsErrorEnabled
       ALIAS UnreferencedVariable
       ALIAS UninitializedVarWarning
       ALIAS WholeProgramOptimization
@@ -800,7 +800,7 @@ foreach(_compilerFlagName IN ITEMS "CMAK
         ExceptionsEnabled
         MultiProcessorCompilation
         DeadCodeEliminate
-        TreatWarnAsErrorEnabled
+        #TreatWarnAsErrorEnabled
   )
 
   if(IGC_OPTION__UNIVERSAL_DRIVER)
